contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -42,13 +43,27 @@ procedure Lkt_Toolbox is
 
    use Liblktlang;
 
+   package Invalid_Decl_Maps is new Ada.Containers.Hashed_Maps
+     (Key_Type        => Analysis.LK_Node,
+      Element_Type    => Boolean,
+      Hash            => Analysis.Hash,
+      Equivalent_Keys => "=");
+
    procedure Print_Semantic_Result
      (S : Analysis.Semantic_Result; Unit : Analysis.Analysis_Unit);
    --  Print a semantic result
 
    function Format_Node (Decl_Node : Decl'Class) return String;
    --  Format node for semantic result printing
 
+   function Populate_Invalid_Decl_Map
+      (Node : Analysis.LK_Node'Class) return Common.Visit_Status;
+   --  Populate invalid declarations map and check for nested invalid
```

Comment:
I would suggest to directly mention the map variable that this populates, and be more specific “invalid declaration” is a bit confusing: a decl with the @invalid annotation versus a declaration that is actually invalid: we can’t assume there that they are the same, otherwise lkt_toolbox would work only on correct tests, while the point here is to check test correctness.
```
Populate ``Invalid_Decl_Map`` and reject declarations with ``@invalid``
annotations that are nested in another declaration annotated with ``@invalid``.
```